src/service-module/types.ts

```diff
@@ -59,3 +64,268 @@ export interface MakeServicePluginOptions {
   mutations?: {}
   actions?: {}
 }
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface FeathersVuexStoreState {
+  /** Allow clients to augment store state */
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface FeathersVuexGlobalModels {
+  /** Allow clients to augment Global models */
+}
+
+// Alias and default to any if user doesn't augment interfaces
+export type StoreState = keyof FeathersVuexStoreState extends never ? any : FeathersVuexStoreState
+export type GlobalModels = keyof FeathersVuexGlobalModels extends never ? any : FeathersVuexGlobalModels
+
+export interface PatchParams<D extends {} = AnyData> extends Params {
+  data?: Partial<D>
+}
+
+export interface ModelSetupContext {
+  /**
+   * The global Vuex store
+   */
+  store: StoreState
+  /**
+   * The global `models` object
+   */
+  models: GlobalModels
+}
+
+export interface ModelInstanceOptions {
+  /**
+   * Creating clone?
+   *
+   * Default: `false`
+   */
+  clone?: boolean
+  /**
+   * Add to store
+   *
+   * Default: `true`
+   */
+  commit?: boolean
+  /**
+   * Merge with existing?
+   *
+   * Default: `true`
+   */
+  merge?: boolean
+}
+
+export type AnyData = { [key: string]: any }
+
+/** Static Model interface */
+export interface ModelStatic extends EventEmitter {
```

Comment:
I've marked static props `readonly` but noticed while adding documentation that [adding static props to the class via the dot operator](https://vuex.feathersjs.com/model-classes.html#extending-the-basemodel-class) is documented and therefore probably should stay supported.

Should I make these props mutable or force clients to override with `//@ts-ignore`?

Reply:
@hamiltoes Let's leave them mutable for now.